### PR TITLE
fix: extra context info not adding to messages

### PR DESCRIPTION
**Pull request by BlackAmda**
Can't add extra context info into messages.
Like, forwardingScore, isForwarded, externalAdReply etc...

Example: 
```js
await sock.sendMessage(id, { text: 'oh hello there', contextInfo: { forwardingScore: 1, isForwarded: true }})
// some random context info (can show a forwarded message)
```

Fix: #2161  #1747

### CODE REVIEW

**Comment by BlackAmda** [comment]
+1

**Comment by BlackAmda** [comment]
+1

**Comment by BlackAmda** [comment]
> This PR is stale because it has been open 6 days with no activity. Remove the stale label or comment or this will be closed in 2 days

+1

File: src/Utils/messages.ts [hunk 1]
```diff
@@ -454,6 +454,11 @@ export const generateWAMessageContent = async(
 		m[messageType].contextInfo.mentionedJid = message.mentions
 	}
 
+	if('contextInfo' in message && message) {
```
Comment (adiwajshing) [inline src/Utils/messages.ts:457]: contextInfo could be undefined

File: src/Utils/messages.ts [hunk 2]
```diff
@@ -454,6 +454,11 @@ export const generateWAMessageContent = async(
...
+		const [messageType] = Object.keys(m)
+		m[messageType].contextInfo = m[messageType] || { }
```
Comment (adiwajshing) [inline src/Utils/messages.ts:459]: merge this contextInfo into the existing one if it exists

File: src/Utils/messages.ts [hunk 3]
```diff
@@ -454,6 +454,11 @@ export const generateWAMessageContent = async(
 		m[messageType].contextInfo.mentionedJid = message.mentions
 	}
 
+	if('contextInfo' in message && message) {
+		const [messageType] = Object.keys(m)
```
Comment (adiwajshing) [inline src/Utils/messages.ts:458]: messageType is defined before this, use it from there -- no need to define the same variable twice

**Pull request closed without merging**